core/server/controllers/frontend/index.js

```diff
@@ -67,7 +67,10 @@ frontendControllers = {
             }
 
             // CASE: we only support /:slug format for pages
-            if (post.page && post.url !== req.path) {
+            // However, given that a url with `/edit/` appended is different
+            // than the post url, we must remove `/edit/` from the url if it
+            // is an edit url.
+            if (post.page && post.url !== ((lookup.isEditURL) ? req.path.replace(/edit\/$/, '') : req.path)) {
```

Comment:
Yes so true, so i think the problem sits in `post-lookup.js`, because it should only detect `isEditURL` if static page has no date permalink.